Pg eng refactor controller UI display

nrice25 opened the pull request:
[Jira Link](https://bradleycapstone.atlassian.net/browse/LV-2268?atlOrigin=eyJpIjoiNjk0MzBjNmJlMmUyNDQ1ZTllMzQxZmU2MWRlNmZjOTciLCJwIjoiaiJ9)

This makes it so that UI should swap automatically based on whether or not the player has used a controller or keyboard last.

Testing time: 20 minutes est.

(The code is messy but it's 1 am and I have to wake up at 6:30 am)

nrice25 committed:
Unpolished V1 UI auto swap

The Main menu only partially works. The cutscenes don't work yet.
Gameplay works.

nrice25 committed:
Merge branch 'development-post-grad' into pg-eng-refactor-controller-ui-display

nrice25 committed:
Merge branch 'development-post-grad' into pg-eng-refactor-controller-ui-display

nrice25 committed:
It works

This finally makes the UI swap automatically

nrice25 committed:
Forgot to delete old setting

RyanASwanson commented on Assets/Scripts/Managers/UniversalManagers/MainUniversalManagers/ControlsManager.cs:
```diff
@@ -23,6 +25,21 @@ public class ControlsManager : MainUniversalManagerFramework
+        if (!Gamepad.current.IsUnityNull())
+        {
+            // We have 0 input users at the start...
+            Debug.Log(InputUser.all.Count);
```
We can delete this debug

RyanASwanson commented on Assets/Scripts/Managers/UniversalManagers/MainUniversalManagers/ControlsManager.cs:
```diff
@@ -23,6 +25,21 @@ public class ControlsManager : MainUniversalManagerFramework
+        }
+        else
+        {
+            Debug.Log("Whos asking??");
```
This guy too

RyanASwanson commented on Assets/Scripts/Managers/UniversalManagers/MainUniversalManagers/ControlsManager.cs:
```diff
@@ -31,14 +48,18 @@ public class ControlsManager : MainUniversalManagerFramework
     /// <param name="device">The device that was changed to</param>
     private void ControlSwap(InputUser user, InputUserChange userChange, InputDevice device)
     {
+        //Debug.Log(device);
```
Also this lad

RyanASwanson commented on Assets/Scripts/Managers/UniversalManagers/MainUniversalManagers/ControlsManager.cs:
```diff
@@ -31,14 +48,18 @@ public class ControlsManager : MainUniversalManagerFramework
         {
             return;
         }
+        Debug.Log(device + " " + userChange);
```
This fine gentleman can also go

RyanASwanson commented on Assets/Scripts/Managers/UniversalManagers/MainUniversalManagers/UiManager.cs:
```diff
@@ -117,16 +119,25 @@ public void SelectUiOnPreviousPage()
+        
         SaveManager.Instance.GetGameSaveData().IsUsingController = _isUsingController;
         ToggleMouse();
+        Debug.Log("crayon eating " + _isUsingController);
```
No I understand this one. I resonate with it. But it can still go

RyanASwanson commented on Assets/Scripts/Managers/UniversalManagers/MainUniversalManagers/UiManager.cs:
```diff
@@ -143,14 +154,23 @@ private void ToggleMouse()
+    protected override void SubscribeToEvents()
+    {
+        base.SubscribeToEvents();
+        ControlsManager.Instance.GetOnChangeControls.AddListener(SwapInput);
```
I don't see this getting unsubscribed, but this is a universal manager so there is no reason it really ever should I suppose. I dunno, I feel it's good practice to add an unsubscribe, but it really isn't mandatory.

nrice25 replied:
Added the listener to OnDisable()

RyanASwanson commented on Assets/Scripts/MenuFunctionality/TitleScreenScrolling.cs:
```diff
@@ -134,6 +137,7 @@ private void PlayMainMenuSplash()
     /// </summary>
     public void OnUiSwap()
     {
+        Debug.Log("GRravyds " + UiManager.IsUsingController);
```
Classic 1am coding debugs. "GRravyds"

RyanASwanson commented on Assets/Scripts/PlayerScripts/PlayerCameraController.cs:
```diff
@@ -628,17 +628,17 @@ private void ChangeCameraAccelTime(InputDevice device)
-            _cinemachinePOV.m_HorizontalAxis.m_AccelTime = _horizontalAccelTimeMK;
-            _cinemachinePOV.m_VerticalAxis.m_AccelTime = _verticalAccelTimeMK;
+            _cinemachinePOV.m_HorizontalAxis.m_AccelTime = _horizontalAccelTimeController;
+            _cinemachinePOV.m_VerticalAxis.m_AccelTime = _verticalAccelTimeController;
```
No idea how I missed this was backwards initially

nrice25 replied:
It worked backwards because we were sending events whenever a controller or M+K would be disconnected. I added another filter to make it so that it isn't reversed anymore.

RyanASwanson reviewed (approved):
Mostly just clean up work to do. The UI on the tutorial and note to change page doesn't seem to change properly, but that's it.

nrice25 committed:
Applied fixes

Removed debugs and the tutorial should now work properly.
Lastly, I made sure UI manager properly removed a listener

nrice25 committed:
Merge branch 'development-post-grad' into pg-eng-refactor-controller-ui-display

nrice25 committed:
The mouse should properly toggle now

Title
I also changed some != null to ! .IsUnityNull()

nrice25 committed:
Merge branch 'development-post-grad' into pg-eng-refactor-controller-ui-display

nrice25 committed:
Merge branch 'development-post-grad' into pg-eng-refactor-controller-ui-display

nrice25 committed:
Merge branch 'development-post-grad' into pg-eng-refactor-controller-ui-display

nrice25 committed:
Merge branch 'development-post-grad' into pg-eng-refactor-controller-ui-display

nrice25 committed:
Fixed merge and steam issue

Added a lot of if statement checks for steam to prevent issues

nrice25 committed:
Merge branch 'development-post-grad' into pg-eng-refactor-controller-ui-display

nrice25 committed:
Cursor bug fixed

The player can now use the cursor once more after hitting pause and then going to the title screen.

RyanASwanson commented on Assets/Scripts/Managers/UniversalManagers/MainUniversalManagers/UiManager.cs:
```diff
@@ -141,18 +141,45 @@ public void SwapInput(InputDevice device)
+        }
+    }
+
+    private void ToggleMouse(bool isPaused, bool hasAudio)
```
This could use an xml.

RyanASwanson reviewed (commented):
I am seeing an Index out of range error on starting the game, but it doesn't seem to be having any effect. Aside from that just one missing XML.

nrice25 committed:
Merge branch 'development-post-grad' into pg-eng-refactor-controller-ui-display

nrice25 committed:
This shouldn't break anything

In fact, this should fix the ability to make a build in my branch.

nrice25 committed:
Update UiManager.cs

nrice25 committed:
UI swap functions again
